kubeflow_batch_predict/dataflow/batch_prediction.py

```diff
@@ -207,7 +211,9 @@ def process(self, element, model_dir):
           loaded_data = [json.loads(d) for d in element]
       else:
         loaded_data = [json.loads(d) for d in element]
-      instances = mlprediction.decode_base64(loaded_data)
+      loaded_data = mlprediction.decode_base64(loaded_data)
+      instances = loaded_data[self._instances_key]
```

Comment:
To make it compatible to the case that there is no "_instances_key", put a if here or an abstract class here. Also, try to "rowfy" the inputs by transforming it to list of 
[{"instance_key": value1, "meta1": xx, "meta2": yy}, {"instance_key": value2, "meta1": xx, "meta2": yy}]. This way, we can do multi-fanout in line 235.